### ISSUE TITLE
Value cannot be null in MixedRealityInputSystem Reset method

### ISSUE THREAD

**Issue by Agredek** [microsoft/MixedRealityToolkit-Unity#5777]
## Describe the bug

ArgumentNullException is being thrown when MixedRealityInputSystem.Reset() method is being called.

Full StackTrace:
>ArgumentNullException: Value cannot be null.
Parameter name: _unity_self
Microsoft.MixedReality.Toolkit.Windows.Input.GestureRecognizerExtensions.UpdateAndResetGestures (UnityEngine.XR.WSA.Input.GestureRecognizer recognizer, UnityEngine.XR.WSA.Input.GestureSettings gestureSettings) (at Assets/MixedRealityToolkit/Extensions/GestureRecognizerExtensions.cs:11)
Microsoft.MixedReality.Toolkit.WindowsMixedReality.Input.WindowsMixedRealityDeviceManager.set_GestureSettings (Microsoft.MixedReality.Toolkit.Windows.Input.WindowsGestureSettings value) (at Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityDeviceManager.cs:178)
Microsoft.MixedReality.Toolkit.WindowsMixedReality.Input.WindowsMixedRealityDeviceManager.Enable () (at Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityDeviceManager.cs:271)
Microsoft.MixedReality.Toolkit.MixedRealityToolkit.RegisterServiceInternal (System.Type interfaceType, Microsoft.MixedReality.Toolkit.IMixedRealityService serviceInstance) (at Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs:834)
Microsoft.MixedReality.Toolkit.MixedRealityToolkit.RegisterServiceInternal[T] (T serviceInstance) (at Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs:849)
Microsoft.MixedReality.Toolkit.MixedRealityToolkit.RegisterService[T] (System.Type concreteType, Microsoft.MixedReality.Toolkit.Utilities.SupportedPlatforms supportedPlatforms, System.Object[] args) (at Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs:206)
Microsoft.MixedReality.Toolkit.MixedRealityToolkit.RegisterDataProvider[T] (System.Type concreteType, Microsoft.MixedReality.Toolkit.Utilities.SupportedPlatforms supportedPlatforms, System.Object[] args) (at Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs:301)
Microsoft.MixedReality.Toolkit.Input.MixedRealityInputSystem.Enable () (at Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputSystem.cs:221)
Microsoft.MixedReality.Toolkit.Input.MixedRealityInputSystem.Reset () (at Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputSystem.cs:239)
API.Utils.VoiceCommandsHelper.AddVoiceCommands (System.String label, System.String alternative) (at Assets/API/Utils/VoiceCommandsHelper.cs:39)

## To reproduce

You can use any scene to check this bug. I've used my own scene and also tried it on Speech Demo scene to make sure that nothing from my own code is interfering.

1. Allow MixedRealityToolkit to initialize
2. Call Reset() method for MixedRealityInputSystem

## Expected behavior

GestureRecognizer should not be null when Reset() method reaches Enable().

## Your Setup (please complete the following information)

- Unity Version
2018.3.14f1
- MRTK Version
MRTK 2.0 RC2.1

## Target Platform (please complete the following information)

- HoloLens
- HoloLens 2

## Additional context

I'm trying to add Voice Commands to the MRTK profile in runtime. While adding new keywords wasn't hard (with the use of reflection), they do not work until next startup. While searching for a way to call some kind of registration of the voice commands, I stumbled upon Reset method in IMixedRealityService. Because of this bug I'm not sure if it will help me, but since I've found it I wanted to report it.

**Comment by Agredek** [comment]
> @Agredek thanks for reporting this!
> 
> There are a known class of bugs around MRTK object shutdown/restart (i.e. see #5085) for another example. This is something we're looking at this iteration.

Thanks for the update, @wiwei!

**Comment by wiwei** [comment]
@Agredek FYI, I'm not sure that I'm going to be able to push in my change for the current iteration because I'm going to be out for a little bit and won't be able to get to the migration tooling before then. Because my change is breaking (in that a thing that is a MonoBehaviour is no longer one), I think that providing the migration tooling is SUPER important before going out.

That being said, I wanted to send out the change so that folks who were interested in grabbing that functionality could unblock themselves - the overall shape of this *shouldn't* change that much.

**Comment by wiwei** [comment]
@Agredek, can you share a little more on your scenario in which you're using the Reset() or Disable/Enable functionality? Some folks have asked some questions regarding the right way of addressing this and I want to make sure that the linked PR actually does address the scenario you're trying to accomplish.

**Comment by Agredek** [comment]
@wiwei
Hi! Sorry for late response. I've been out on holidays.

Since submiting this issue I've changed my approach to this problem, but if I recall correctly I wanted to add Voice Commands to my dynamically created buttons. Basically I wanted to register all the voice commands on the start and disable them when buttons are removed form the scene. While working on the topic I've discovered that MRTK probably does some work on Awake and adding voice commands this way won't work.
